### PR TITLE
MONGOID-5293 Add Rails-style defaults & config.load_defaults

### PR DESCRIPTION
**Pull request by Neilshweky**
MONGOID-5293

### CODE REVIEW

File: lib/mongoid/config/defaults.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        # Note that for 7.x, since all of the feature flag defaults have been
+        # flipped to the new functionality, all of the settings for those
+        # versions are to give old functionality. Because of this, it is
+        # possible to recurse to later version to get all of the options to
+        # turn off. Note that this won't be true when adding feature flags to
+        # 8.1, since the default will be the old functionality until the next
+        # major version is released. More likely, the recursion will have to go
+        # in the other direction (towards earlier versions).
+
+        case version.to_s
```
Comment (p-mongo) [inline lib/mongoid/config/defaults.rb:26]: This should explicitly round to 1 decimal place before the `to_s` call for good measure.

Or maybe more than one but there should be some sort of rounding to a certain precision performed.

File: lib/mongoid/config/defaults.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        when "7.5"
+          # flags introduced in 8.0 - old functionality
+          Mongoid.legacy_attributes = true
+          Mongoid.map_big_decimal_to_decimal128 = false
+        when "8.0"
+          # All flag defaults currently reflect 8.0 behavior.
+        when "8.1"
+          # All flag defaults currently reflect 8.1 behavior.
+        else
+          raise ArgumentError, "Unknown version: #{version.to_s}"
```
Comment (p-mongo) [inline lib/mongoid/config/defaults.rb:54]: `to_s` is not needed when the value is already inside a string being interpolated.

File: lib/mongoid/config/defaults.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+        case version.to_s
+        when "7.3"
+          # flags introduced in 7.4 - old functionality
+          Mongoid.broken_aggregables = true
+          Mongoid.broken_alias_handling = true
+          Mongoid.broken_and = true
+          Mongoid.broken_scoping = true
+          Mongoid.broken_updates = true
+          Mongoid.compare_time_by_ms = false
```
Comment (p-mongo) [inline lib/mongoid/config/defaults.rb:34]: Is `self` here the `Mongoid` module?

**Review by p-mongo — COMMENTED** [review]
Can you please add the user documentation for this feature?

File: lib/mongoid/config/defaults.rb (hunk 1 above)
Comment (Neilshweky) [reply lib/mongoid/config/defaults.rb:26]: I followed Rails' lead on this one... The parameter can also be a string and it feels awkward to convert to a number, round, and convert back to a string.

File: lib/mongoid/config/defaults.rb (hunk 2 above)
Comment (Neilshweky) [reply lib/mongoid/config/defaults.rb:54]: done

File: lib/mongoid/config/defaults.rb (hunk 3 above)
Comment (Neilshweky) [reply lib/mongoid/config/defaults.rb]: yes, done.

File: lib/mongoid/config/defaults.rb (hunk 1 above)
Comment (p-mongo) [reply lib/mongoid/config/defaults.rb:26]: If that is how Rails does it it can't be wrong.

**Review by p-mongo — APPROVED** [review]
I approve what is here but this needs user documentation on https://www.mongodb.com/docs/mongoid/master/reference/configuration/ giving  a usage example and ideally listing the option bundles that are defined.

**Review by p-mongo — APPROVED** [review]
:+1: Very nice.

**Comment by johnnyshields** [comment]
Neil, great work on this, it makes configuration much, much simpler 🎉  I've raised https://jira.mongodb.org/browse/MONGOID-5439 as a follow-up.

**Pull request merged**